src/Docs/Resources/current/4-how-to/670-administration-language-switcher.md

```diff
@@ -0,0 +1,256 @@
+
+```twig
+{# foobar-detail.html.twig #}
+
+<sw-page>
+    <template slot="language-switch">
+        <sw-language-switch></sw-language-switch>
+    </template>
+</sw-page>
+```
+
+Like in the listing you have to provide a reloading functionality by listening to the on-change event:
+
+```twig
+<sw-language-switch v-on:on-change="loadItem"></sw-language-switch>
+``` 
+
+```javascript
+import template from './foobar-detail.html.twig';
+
+Shopware.Component.register('foobar-detail', {
+    template,
+
+    inject: [
+        'repositoryFactory',
+        'context'
+    ],
+
+    data() {
+        return {
+            item: null
+        };
+    },
+
+    created() {
+        this.loadItem();
+    },
+
+    computed: {
+        customEntityRepository() {
+            return this.repositoryFactory.create('foobar_custom_entity');
+        }
+    },
+
+    methods: {
+        loadItem() {
+            return this.customEntityRepository.get(this.$route.params.id, this.context).then(entity => {
+                this.item = entity
+            });
+        }
+    }
+});
+```
+
+This will now reload your entity on a language switch.
+But be aware of that this might end up in data loss.
+To prevent that the language switch component can cancel the language switch if you provide callbacks that check for changes:
+
+```twig
+<sw-language-switch v-bind:saveChangesFunction="saveOnLanguageChange"
+                    v-bind:abortChangeFunction="abortOnLanguageChange"
+                    v-on:on-change="loadItem">
+</sw-language-switch>
+``` 
+
+```javascript
+import template from './foobar-detail.html.twig';
+
+Shopware.Component.register('foobar-detail', {
+    template,
+
+    inject: [
+        'repositoryFactory',
+        'context'
+    ],
+
+    data() {
+        return {
+            item: null
+        };
+    },
+
+    created() {
+        this.loadItem();
+    },
+
+    computed: {
+        customEntityRepository() {
+            return this.repositoryFactory.create('foobar_custom_entity');
+        }
+    },
+
+    methods: {
+        loadItem() {
+            return this.customEntityRepository.get(this.$route.params.id, this.context).then(entity => {
+                this.item = entity
+            })
+        },
+
+        abortOnLanguageChange() {
+            return this.customEntityRepository.hasChanges(this.item)
+        },
+
+        saveOnLanguageChange() {
+            return this.customEntityRepository.save(this.item, this.context)
+        }
+    }
+});
+```
+
+It is regarded a good way to communicate the current selected language as different values are displayed than usual.
+For that you can simply use the language info component.
+This component displays a short info text about the current selected language and its impacts on the current entity.
+You can see it regularly above an other content:
+
+```twig
+<sw-page class="foobar-detail">
+    <sw-card-view slot="content"
```

Comment:
Please move the slot from card-view to a surrounding `<template #content>` element.

Reply:
I want to provide a minimal example of code. A user should decide on their own whether they want to wrap things up in a pseudo-container. Are there any downsides without `template` when I have a single element for this slot?

Reply:
I understand that! I may be wrong, but afaik it's not possible to use slots directly on the component in Vue3, so it'd be better to show a future safe way of implementing.

Reply:
I didn't play with vue3 yet. I can compare and see it there will be possible upgrade problems.